gatk_merge_sam_files_4.1.8.0/gatk_merge_sam_files_4.1.8.0.cwl

```diff
@@ -0,0 +1,210 @@
+  - id: sort_order
+    type: string?
+    inputBinding:
+      position: 1
+      prefix: --SORT_ORDER
+    doc: |
+      Sort order of output file  Default value: coordinate. Possible values: {unsorted,
+      queryname, coordinate, duplicate, unknown}
+  - id: use_threading
+    type: boolean?
+    inputBinding:
+      position: 1
+      prefix: --USE_THREADING
+    doc: |
+      Option to create a background thread to encode, compress and write to disk the output
+      file. The threaded version uses about 20% more CPU and decreases runtime by ~20% when
+      writing out a compressed BAM file.  Default value: false. Possible values: {true, false}
+  - id: validation_stringency
+    type: string?
+    inputBinding:
+      position: 1
+      prefix: --VALIDATION_STRINGENCY
+    doc: |
+      Validation stringency for all SAM files read by this program.  Setting stringency to
+      SILENT can improve performance when processing a BAM file in which variable-length data
+      (read, qualities, tags) do not otherwise need to be decoded.  Default value: STRICT.
+      Possible values: {STRICT, LENIENT, SILENT}
+  - id: verbosity
+    type: string?
+    inputBinding:
+      position: 1
+      prefix: --VERBOSITY
+    doc: |
+      Control verbosity of logging.  Default value: INFO. Possible values: {ERROR, WARNING,
+      INFO, DEBUG}
+outputs:
+  - id: gatk_merge_sam_files_output
+    type: File
+    outputBinding:
+      glob: |-
+        ${
+            if(inputs.output_file_name){
+                return inputs.output_file_name
+            } else {
+                return 'merged.bam'
+            }
+        }
+arguments:
+  - position: 0
+    prefix: '--java-options'
+    valueFrom: |-
+      ${
+        if(inputs.memory_per_job && inputs.memory_overhead) {
+          if(inputs.memory_per_job % 1000 == 0) {
+            return "-Xmx" + (inputs.memory_per_job/1000).toString() + "G"
+          }
+          else {
+            return "-Xmx" + Math.floor((inputs.memory_per_job/1000)).toString() + "G"
+          }
+        }
+        else if (inputs.memory_per_job && !inputs.memory_overhead){
+          if(inputs.memory_per_job % 1000 == 0) {
+            return "-Xmx" + (inputs.memory_per_job/1000).toString() + "G"
+          }
+          else {
+            return "-Xmx" + Math.floor((inputs.memory_per_job/1000)).toString() + "G"
+          }
+        }
+        else if(!inputs.memory_per_job && inputs.memory_overhead){
+          return "-Xmx15G"
+        }
+        else {
+            return "-Xmx15G"
+        }
+      }
+  - position: 0
+    prefix: '--TMP_DIR'
+    valueFrom: .
+  - position: 0
+    prefix: '--COMPRESSION_LEVEL'
+    valueFrom: '2'
+  - position: 0
+    prefix: '--MAX_RECORDS_IN_RAM'
+    valueFrom: '50000'
+  - position: 2
+    prefix: '-O'
+    valueFrom: |-
+      ${
+          if(inputs.output_file_name){
+              return inputs.output_file_name
+          } else {
+              return 'merged.bam'
```

Comment:
Can we use the input bam name as prefix here if the the output file name is not given and have it as `_merged.bam`

Reply:
- should we have the output as sam or bam?
- I also noticed that we have two tools where we name the output as `_merged.bam` not sure if it matters but just wanted to point, whether we should have unique names.

Reply:
@rhshah I left it is `merged.bam` because this tool can accept multiple inputs, so it is not clear which one to use as the base name. Although I am not sure how exactly this tool is being used in the pipeline, I figured the tool will be most generalizable with how it is now